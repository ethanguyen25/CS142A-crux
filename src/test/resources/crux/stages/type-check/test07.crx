// test local variable

void main() {
  int a;
  a = 42;
  printInt(a);
}
