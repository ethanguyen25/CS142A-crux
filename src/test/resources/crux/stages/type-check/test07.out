Crux Program has no type errors.
ast.DeclarationList(3)
  ast.FunctionDefinition(3)[Symbol(main:func(TypeList()):void), []]
    ast.StatementList(4)
      ast.VariableDeclaration(4)[Symbol(a:int)]
      void: ast.Assignment(5)
        Address(int): ast.Name(5)[a]
        int: ast.LiteralInt(5)[42]
      void: ast.Call(6)[Symbol(printInt)]
        int: ast.Dereference(6)
          Address(int): ast.Name(6)[a]
